align_anything/evaluation/benchmarks/AGIEval/vllm_eval.py

```diff
@@ -0,0 +1,245 @@
+        update_results(output_dir, brief_filename, detailed_filename,task2details)
+        
+        return task2details
+    
+def evaluator(raw_output: List[InferenceOutput], dataloader: AGIEvalDataLoader, task: str):
+    dataset = load_dataset(dataloader.task_dir, task)[dataloader.split]
+    correct_answers = []
+    responses = []
+    true_cases = []
+    false_cases = []
+    cnt_sum = 0
+    cnt_match = 0
+    cnt_fail = 0
+    flag_fail = True
+    for instance in dataset:
+        correct_answers.append(
+            {
+                'prompt': instance['question'],
+                'choices': instance['choices'],
+                'answer': dataloader.get_answer(instance)
+            }
+        )
+    for item in raw_output:
+        dataloader.candidate_labels = get_candidate_labels(item.prompt)
+        responses.append(
+            {
+                'prompt': (item.prompt),
+                'answer_logprobs': get_chosen_answer(item.response_logprobs[-1], dataloader.candidate_labels),
+                'answer': item.response[0]
+            }
+        )
+    for correct_answer in correct_answers:
+        cnt_sum += 1
+        for response in responses:
+            if correct_answer['prompt'] in response['prompt']:
+                flag_fail = False
+                chosen_answer = max(response['answer_logprobs'], key=response['answer_logprobs'].get)
+                eval_case = {
+                    'question': correct_answer['prompt'],
+                    'choices': correct_answer['choices'],
+                    'correct_answer': correct_answer['answer'],
+                    'answer_logprobs': response['answer_logprobs'],
+                    'chosen_answer': chosen_answer
+                }
+                if judge_answer(correct_answer['answer'], chosen_answer, response['answer']):
+                    cnt_match += 1
+                    eval_case['result'] = True
+                    true_cases.append(eval_case)
+                else:
+                    eval_case['result'] = False
+                    false_cases.append(eval_case)
+                break
+        if flag_fail:
+            cnt_fail += 1
+        else:
+            flag_fail = True
+        
+    return cnt_match, cnt_sum, true_cases, false_cases
+
+def get_chosen_answer(logprobs: List[Dict[str, Any]], candidate_answers: List[str]):
+    answer_logprobs = {}
+    for logprob in logprobs:
+        key = next(iter(logprob.values())).decoded_token
+        value = next(iter(logprob.values())).logprob
+        if key in candidate_answers:
+            answer_logprobs[key] = value
+    for label in candidate_answers:
+        if label not in answer_logprobs.keys():
+            answer_logprobs[label] = -100.0
+            
+    return answer_logprobs
+
+def judge_answer(correct_answer: str, chosen_answer: str, answer: str):
+    if correct_answer.lower() == answer.lower() or correct_answer.lower() == chosen_answer.lower():
```

Comment:
You mean all tasks? Some benchmarks like CEval has more than 50 tasks, I have to test it one by one?

Reply:
Instead of testing each task individually, you can run tests on all tasks at once. If any bugs arise, we can then focus on those specific tasks and make adjustments as needed. This approach might save time and still ensure comprehensive coverage.

Reply:
In one benchmark, if there's different data style, why shouldn't we modify the data itself or simply leave it.

Reply:
We used the data from Hugging Face for the tests, so we had to adapt our code to match the format of that data. Modifying the data will be the next step. Additionally, omitting some data could result in the model missing certain aspects of the evaluation.

Reply:
> In one benchmark, if there's different data style, why shouldn't we modify the data itself or simply leave it.

If you modify the benchmark, you should give us your modified data and prove  that the change won't affect the final validation result

Reply:
So, you didn't notice originally there's no AGIEval dataset for all tasks on HF?

Reply:
> So, you didn't notice originally there's no AGIEval dataset for all tasks on HF?

We have noticed this question, so you can send your modified benchmark to us with a readme proving your modified benchmark containing all the test data, we can help you upload the modified benchmark to huggingface. In future work, if there is no hf version benchmark, you can still send the modified benchmark to us. But the most important things is that the modification does not effect the final validation result

Reply:
Some of the tasks that are available on HF didn't even pass. So, you didn't notice that?

Reply:
> Some of the tasks that are available on HF didn't even pass. So, you didn't notice that?

It seems like you could check it by yourself and find out hidden bugs, so why don't you directly modify it?

Reply:
The code review and ensuring its correctness is a shared responsibility, especially when it comes to code you've written. If there are bugs, it would be most efficient for the original author to fix them, as you have the best understanding of the implementation. Let's focus on resolving the problems together to ensure everything works as expected. OK?

Reply:
You mean "**_together_**"? What kind of help you offer? As the main author of the repo, I write the code you provided to fit the benchmark, it should be similar and familiar to you. Why you always simply say you need to do what, what, what, what you help? I provided my code for your repo, but we're not in a superior-subordinate relationship, right? This isn't my obligation, is it?